src/data_switch.rs

```diff
@@ -82,17 +82,26 @@ pub struct GeoPoint {
 /// represented by its vertices as a sequence of lat-lon points
 pub type Polygon = [GeoPoint];
 
-/// Container of series data
-#[derive(Debug, Clone, PartialEq)]
-pub struct SeriesCache {
+/// Data container
+///
+/// a [`new`](DataCache::new) method is provided to
+/// avoid the need to construct an R*-tree manually
+// TODO: check why we needed to derive PartialEq for SeriesCache
+#[derive(Debug, Clone)]
+pub struct DataCache {
+    /// Vector of timeseries in chronological order.
+    /// The timeseries data can originate from a single station or from
+    /// a collection of neighboring stations
```

Comment:
I would remove this part. The stations don't actually have to be neighbouring, in fact we expect them in practice to cover large areas (potentially the whole globe), as the spatial tests are smart enough to figure out which stations in the set neighbour each other. As for the "single station" part, that's just a degenerate case, and I don't think warrants a special mention